framework/src/meshgenerators/MeshDiagnosticsGenerator.C

```diff
@@ -266,6 +278,69 @@ MeshDiagnosticsGenerator::checkSidesetsOrientation(const std::unique_ptr<MeshBas
   }
 }
 
+void
+MeshDiagnosticsGenerator::checkWaterTightSidesets(const std::unique_ptr<MeshBase> & mesh) const
+{
+  /*
+  Algorithm Overview:
+  1) Loop through all elements
+  2) For each element loop through it's edges
+  3) If !neighbors it's external
+  4) If external check if it's part of a sideset
+  */
+  if (mesh->mesh_dimension() < 2)
+    mooseError("The sideset check only works for 2D and 3D meshes");
+  if (!mesh->is_serial())
+    mooseError("Only serialized/replicated meshes are supported");
+  auto & boundary_info = mesh->get_boundary_info();
+  boundary_info.build_side_list();
+  const auto sideset_map = boundary_info.get_sideset_map();
+  unsigned int num_faces_without_sideset = 0;
+
+  for (const auto elem : mesh->active_element_ptr_range())
+  {
+    std::vector<std::unique_ptr<Elem>> elem_sides(elem->n_sides());
+    for (auto i : elem->side_index_range())
+    {
+      // Check if side is external
+      if (elem->neighbor_ptr(i) == nullptr)
+      {
+        // If external check if that side had a sideset
+        auto side_itr = sideset_map.equal_range(elem);
```

Comment:
```suggestion
        auto side_range = sideset_map.equal_range(elem);
```
itr is fine for an iterator